lib/src/migrators/module.dart

```diff
@@ -578,17 +678,23 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     if (node.span.sourceUrl == _currentUrl) return null;
     if (!_namespaces.containsKey(node.span.sourceUrl)) {
       /// Add new use rule for indirect dependency
-      var relativePath = p.relative(node.span.sourceUrl.path,
-          from: p.dirname(_currentUrl.path));
-      var basename = p.basenameWithoutExtension(relativePath);
-      if (basename.startsWith('_')) basename = basename.substring(1);
-      var simplePath = p.relative(p.join(p.dirname(relativePath), basename));
+      var simplePath = _absoluteUriToDependency(node.span.sourceUrl);
       _additionalUseRules.add(simplePath);
       _namespaces[node.span.sourceUrl] = namespaceForPath(simplePath);
     }
     return _namespaces[node.span.sourceUrl];
   }
 
+  /// Converts an absolute URI for a stylesheet into the simplest string that
```

Comment:
```suggestion
  /// Converts an absolute URL for a stylesheet into the simplest string that
```

Also in the function name. This is kind of a personal crusade for me, but according to [the whatwg spec](https://url.spec.whatwg.org/#goals) "URL" should be universally preferred over "URI" (contrary to Dart's unfortunate choice of class name).